crates/node/src/main.rs

```diff
@@ -152,52 +160,29 @@ impl mempool::Storage for storage::Database {
 
 #[async_trait]
 impl workflow::TransactionStore for storage::Database {
-    async fn find_transaction(&self, tx_hash: &Hash) -> Result<Option<Transaction>> {
+    async fn find_transaction(&self, tx_hash: &Hash) -> Result<Option<Transaction<TxValdiated>>> {
         self.find_transaction(tx_hash).await
     }
 }
 
-struct P2PTxHandler {
-    mempool: Arc<RwLock<Mempool>>,
-    database: Arc<Database>,
-}
-
-impl P2PTxHandler {
-    pub fn new(mempool: Arc<RwLock<Mempool>>, database: Arc<Database>) -> Self {
-        Self { mempool, database }
-    }
-}
-
-#[async_trait::async_trait]
-impl networking::p2p::TxHandler for P2PTxHandler {
-    async fn recv_tx(&self, tx: Transaction) -> Result<()> {
-        // The transaction was received from P2P network so we can consider it
-        // propagated at this point.
-        let tx_hash = tx.hash;
-        let mut tx = tx;
-        tx.propagated = true;
+async fn run(config: Arc<Config>) -> Result<()> {
+    let database = Arc::new(Database::new(&config.db_url).await?);
 
-        // Submit the tx to mempool.
-        self.mempool.write().await.add(tx).await?;
+    let http_peer_list: Arc<tokio::sync::RwLock<HashMap<SocketAddr, Option<u16>>>> =
+        Default::default();
 
-        //TODO copy paste of the asset manager handle_transaction method.
-        //added because when a tx arrive from the p2p asset are not added.
-        //should be done in a better way.
-        self.database.add_asset(&tx_hash).await
-    }
-}
+    let mempool = Arc::new(RwLock::new(Mempool::new(database.clone()).await?));
 
-#[async_trait::async_trait]
-impl mempool::AclWhitelist for Database {
-    async fn contains(&self, key: &PublicKey) -> Result<bool> {
-        let key = entity::PublicKey(*key);
-        self.acl_whitelist_has(&key).await
-    }
-}
-
-async fn run(config: Arc<Config>) -> Result<()> {
-    let database = Arc::new(Database::new(&config.db_url).await?);
-    let file_storage = Arc::new(storage::File::new(&config.data_directory));
+    //start Tx process event loop
+    let (txevent_loop_jh, tx_sender, p2p_stream) = txvalidation::start_event_loop(
+        config.data_directory.clone(),
+        config.p2p_listen_addr,
+        config.http_download_port,
```

Comment:
Initially, I wouldn't add a dep between the txvalidation module and the cli one. There's some issue with module declaration that are declared in main.rs and other in lib.rs. At the beginning, txvalidation was declared in lib and I move it to main because of this issue. I think only module related to main like cli should be declared in main and the other in lib.
Main config and module config are different. Here, I can define a txvalidation::config struct to embbed the config data.

Reply:
>  I think only module related to main like cli should be declared in main and the other in lib.
In general yeah, but right now when both are in node crate, it doesn't make sense to expose the whole node implementation in lib.

But in the near future, I think we should split the crate and pull the core types out of this so that the separation between node bin and lib is clearer.

Reply:
Yes, it's a good idea. I think that if it's the plan, short term we should put all module that will move into lib so that it will easier to move them in another crate then directly from main.